### DISCUSSION TITLE
Discrepancy between dependency tree shown via displacy vs. .children

### DISCUSSION THREAD

**Discussion by fersarr** [explosion/spaCy#5454] [Help: Other Questions] [answered]
## How to reproduce the behaviour
Consider the sentence `please, serve me, please`. For a bit, ignore the fact that the dependency tree seems wrong, since this is an imperative sentence, and performance on them is not great. But note that `Displacy`, using the english model `core_web_sm`, shows it as:

![Screen Shot 2020-05-18 at 5 11 30 PM](https://user-images.githubusercontent.com/3913371/82235923-45af9600-992b-11ea-9024-cdd4616e078b.png)

And when accessing the children via `.children`, there seem to be arrows from the first `please` to the first comma:

```
>>> parsed_sent                                                                                                                                                                 Breakpoints:
please, serve me, please
>>> parsed_sent.root
please
>>> parsed_sent.root.i
0
>>> parsed_sent.root.idx
0
>>> list(parsed_sent.root.children)
[,]
>>> comma = list(parsed_sent.root.children)[0]
>>> comma
,
>>> list(comma.children)
[serve, ,]
```
There is a discrepancy between the tree structure shown by `displacy` and `.children`

## Your Environment

* Operating System: Mac
* Python Version Used: 3.8
* spaCy Version Used:
```
>>> spacy.__version__
'2.2.4'
```

**Comment by adrianeboyd** [comment, marked as answer]
By default displacy simplifies some arcs to make the parse easier to read. See the `collapse_punct` option here: https://spacy.io/api/top-level#displacy_options

**Comment by fersarr** [comment]
aha! I see, sorry I was not aware. Thank you! We can close this.
